### PR TITLE
Migrate Travis CI to Github Actions

### LINKED ISSUE #683: Migrate from Travis to Github Actions
As proposed by @alexellis we should move all our repos to Github Actions because of it's generous free-tier as we're not able to get enough builds out of the free credits that Travis provides.
## Current Behaviour

We're running all our CI builds using Travis CI
## Possible Solution

Move to Github Actions

### PR DESCRIPTION
**Pull request by nitishkumar71**
## Description
We want to migrate from Travis CI to Github Actions
Fixes #683 
openfaas/faas#1585
## How Has This Been Tested?

Please check [CI Only](https://github.com/nitishkumar71/openfaas-cloud/actions/runs/351165720) Build
Please check [Publish](https://github.com/nitishkumar71/openfaas-cloud/actions/runs/355160257) build for tag push

## How are existing users impacted? What migration steps/scripts do we need?
1. We are not able to run CI Pipelines
2. We are moving to github actions for CI Pipelines

## Checklist:

I have:

- [x] updated the documentation and/or roadmap (if required)
- [x] read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] signed-off my commits with `git commit -s`
- [ ] added unit tests

### CODE REVIEW

File: .github/workflows/ci-only.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  push:
+    branches: [ "*" ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        go-version: [1.9.x]
```
Comment (alexellis) [inline .github/workflows/ci-only.yml:13]: This should be 1.13

File: .github/workflows/ci-only.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+name: ci-only
+
+on:
+  push:
+    branches: [ "*" ]
+  pull_request:
+    branches: [ master ]
```
Comment (alexellis) [inline .github/workflows/ci-only.yml:7]: Why only master?

**Review by alexellis — CHANGES_REQUESTED** [review]
I have some suggestions

File: .github/workflows/ci-only.yml (hunk 2 above)
Comment (nitishkumar71) [reply .github/workflows/ci-only.yml:7]: So, should we enable it for both push and pull request?

**Comment by alexellis** [comment]
@wilsonianb could you advise re: pushing tags based upon the release version?

File: .github/workflows/ci-only.yml (hunk 1 above)
Comment (nitishkumar71) [reply .github/workflows/ci-only.yml]: Done

**Comment by wilsonianb** [comment]
You could do something like this (in an `on.tag` workflow) for the non-function components and use `${{ github.ref }}` for the image tags:
https://github.com/wilsonianb/openfaas-cloud/blob/dfc967a49eca9acd099afc02138d96607ea4332b/.github/workflows/docker.yml#L34-L57

**Comment by nitishkumar71** [comment]
> You could do something like this (in an `on.tag` workflow) for the non-function components and use `${{ github.ref }}` for the image tags:
> https://github.com/wilsonianb/openfaas-cloud/blob/dfc967a49eca9acd099afc02138d96607ea4332b/.github/workflows/docker.yml#L34-L57

@wilsonianb Thank you sharing the example.

@alexellis Can you please review it.

**Comment by Waterdrips** [comment]
@nitishkumar71 There's actually a lot more to converting ofc into actions than adding these workflows. As OFC is a complex system with lots of moving parts its probably best for me to take this over.

**Comment by nitishkumar71** [comment]
> @nitishkumar71 There's actually a lot more to converting ofc into actions than adding these workflows. As OFC is a complex system with lots of moving parts its probably best for me to take this over.

makes sense

**Pull request closed without merging**